valkey/templates/_helpers.tpl

```diff
@@ -62,18 +62,60 @@ Create the name of the service account to use
 {{- end }}
 
 {{/*
-Creating Image Pull Secrets
+Returns the Valkey container image
 */}}
-{{- define "imagePullSecret" }}
-{{- with .Values.imageCredentials }}
-{{- printf "{\"auths\":{\"%s\":{\"username\":\"%s\",\"password\":\"%s\",\"email\":\"%s\",\"auth\":\"%s\"}}}" .registry .username .password .email (printf "%s:%s" .username .password | b64enc) | b64enc }}
-{{- end }}
+{{- define "valkey.image" -}}
+{{- $registryName := .Values.image.registry }}
+{{- $repositoryName := .Values.image.repository }}
+{{- $tag := .Values.image.tag | default .Chart.AppVersion }}
+{{- if .Values.global }}
+  {{- if .Values.global.imageRegistry }}
+    {{- $registryName = .Values.global.imageRegistry }}
+  {{- end }}
 {{- end }}
+{{- if $registryName }}
+{{- printf "%s/%s:%s" $registryName $repositoryName $tag }}
+{{- else }}
+{{- printf "%s:%s" $repositoryName $tag }}
+{{ end }}
+{{- end -}}
 
-{{- define "valkey.secretName" -}}
-{{- if .Values.imagePullSecrets.nameOverride }}
-{{- .Values.imagePullSecrets.nameOverride }}
+{{/*
+Returns the Valkey exporter container image
+*/}}
+{{- define "valkey.metrics.exporter.image" -}}
```

Comment:
I do not fancy this almost identical functions, this is a code smell... I will refactor it.